addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BuildingMonitorHandler.java

```diff
@@ -0,0 +1,387 @@
+            logger.debug("Bridge handler not found or not ONLINE.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+
+        } else {
+            if (command instanceof RefreshType) {
+                switch (channelUID.getId()) {
+                    case Wso2iotsBindingConstants.TEMPERATURE:
+                        updateChannel(channelUID.getId(), iotsResponseTemp);
+                        break;
+                    case Wso2iotsBindingConstants.LIGHT:
+                        updateChannel(channelUID.getId(), iotsResponseLight);
+                        break;
+                    case Wso2iotsBindingConstants.MOTION:
+                        updateChannel(channelUID.getId(), iotsResponseMotion);
+                        break;
+                    case Wso2iotsBindingConstants.HUMIDITY:
+                        updateChannel(channelUID.getId(), iotsResponseHumidity);
+                        break;
+                }
+
+            } else {
+                logger.debug("Can not handle command {}", command);
+            }
+        }
+    }
+
+    /**
+     * Start the job refreshing the building monitor data
+     */
+    private void startAutomaticRefresh(BuildingMonitorConfiguration config, BridgeConfiguration configBridge) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                bridgeHandler = getBridgeHandler();
+                if (bridgeHandler == null || !bridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                    logger.debug("Bridge handler not found or not ONLINE.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+
+                } else {
+                    try {
+                        // Request new data
+                        iotsResponseTemp = updateBuildingMonitorData(Wso2iotsBindingConstants.TEMPERATURE, config,
+                                configBridge);
+                        iotsResponseLight = updateBuildingMonitorData(Wso2iotsBindingConstants.LIGHT, config,
+                                configBridge);
+                        iotsResponseMotion = updateBuildingMonitorData(Wso2iotsBindingConstants.MOTION, config,
+                                configBridge);
+                        iotsResponseHumidity = updateBuildingMonitorData(Wso2iotsBindingConstants.HUMIDITY, config,
+                                configBridge);
+
+                        // Update all channels from the updated data
+                        updateChannel(Wso2iotsBindingConstants.TEMPERATURE, iotsResponseTemp);
+                        updateChannel(Wso2iotsBindingConstants.LIGHT, iotsResponseLight);
+                        updateChannel(Wso2iotsBindingConstants.MOTION, iotsResponseMotion);
+                        updateChannel(Wso2iotsBindingConstants.HUMIDITY, iotsResponseHumidity);
+
+                    } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception